src/Model/MethodGo.cs

```diff
@@ -213,6 +214,12 @@ public string MethodReturnSignature(bool helper)
             return $"{retVal}, {errVal}";
         }
 
+        public bool ShouldValidate()
+        {
+            return true == AutoRest.Core.Settings.Instance.Host?.GetValue<bool?>("client-side-validation").Result
+                && !ParameterValidations.IsNullOrEmpty();

```

Comment:
[nit] This is sort of a needlessly long line. I think it would be more readable as:
``` C#
public bool ShouldValidate()
{
    var clientValidation = AutoRest.Core.Settings.Insance.Host?.GetValue<bool?>("client-side-validation").Result;
    return clientValidation && !ParameterValidations.IsNullOrEmpty();
}
```

Preferably, you could reduce down that first line even more :)

Reply:
blame me for the long line, that was my interactive rebasing and adjusting to new project structure on the fly... didn't even check if there's a `using AutoRest.Core;` in that file

Reply:
While we're discussing this chunk of code I don't understand why the "ParameterValidations.IsNullOrEmpty()" clause is necessary.  If the settings that we get from core say "disable validation" isn't that enough?

Reply:
@jhendrixMSFT, I believe that `IsNullOrEmpty()` check still makes sense. It could spare us generating code to allocate what would be an empty slice of validations to enforce, for example.

Reply:
If that's the case then there's more to fix here IMO.  It really should be as simple as "if the flag is set to true then don't emit any validation code" right?

Reply:
Fair enough :) Truthfully, I'm not trying to say what should be, just trying to explain why that call would be there.